pydy/codegen/ode_function_generators.py

```diff
@@ -169,8 +169,8 @@ def _deduce_system_type(**kwargs):
         full mass matrix
             M(x, p) * x' = f(x, t, r, p)
         min mass matrix
-            M(q, p) * u' = f(q, u, t, r, p)
-            q' = g(q, u, t)
+            M(x, p) * x'_d = f(x, t, r, p)
+            x'_k = g(x, t)
```

Comment:
The tests for ode_function_generator tests this case. The use case is any system with which you have the q'=g(x, t) explicitly defined (which is most all systems). For large degree of freedom systems this form should provide much faster evaluation of x' because you only have to invert an n x n matrix instead of an > n x n matrix. We should also support a form like:

M(q, p) u' = f(q, u, t, r, p)
A(q, p) q' = g(q, u, t)

To be more complete, come to think of it.

This last form is the preferred form for computational speed needs.